### PR TITLE
Store temporary reference between document collections and specialist topics

### PR DESCRIPTION
**Pull request by hannako**
We will soon be converting some specialist topics into document collections.
This will be done via a [rake task](https://github.com/alphagov/whitehall/pull/7409) to create a draft document collection.

Storing a reference to the source topic will make it possible to check if a topic
has already been converted, and avoid accidentally creating duplicates.

This is a temporary association, once the conversions have all been completed,
the column can be removed again.

https://trello.com/c/Y8Pf7xFO/1590-rake-task-to-create-a-document-collection-from-a-specialist-topic-m

### CODE REVIEW

**Review by Tetrino — APPROVED** [review]
Looks fine to me!

File: db/schema.rb [hunk 1]
```diff
@@ -93,7 +93,7 @@
     t.integer "consultation_response_form_data_id"
   end
 
-  create_table "contact_number_translations", id: :integer, charset: "utf8mb3", collation: "utf8_unicode_ci", force: :cascade do |t|
+  create_table "contact_number_translations", id: :integer, charset: "utf8mb3", collation: "utf8mb3_unicode_ci", force: :cascade do |t|
```
[the remark this answers is not in the document]
Comment (Tetrino) [reply db/schema.rb:96]: @BeckaL I'm not sure on official strategy, but they're both the same format of 8 byte character support, I believe the new change is just the modern standard of marking it.

**Review by ollietreend — APPROVED** [review]
Looks good to me

File: db/schema.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ollietreend) [reply db/schema.rb]: I've seen similar weirdness with charsets and collations changing – although usually that's when I develop locally vs. in govuk-docker.

So long as the database migration works okay in integration/staging/prod then it doesn't seem too problematic to me.

**Review by BeckaL — APPROVED** [review]
LGTM!

**Pull request merged**